packages/elasticsearch-client/src/suggest.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+
+export type SuggestParameters = {
+  prefix: string;
+  size: number;
+};
+
+export type SuggestExtraParams = {
+  suggestParameters: SuggestParameters;
+};
+
+export type SuggestParams = {
+  client: Client;
+  index: string;
+  suggestParameters: SuggestParameters;
+  searchConfig: any;
+};
+
+export const suggest = async ({
+  client,
+  index,
+  searchConfig,
+  suggestParameters,
+}: SuggestParams) => {
+  const { generateSuggestQuery } = searchConfig;
+
+  const suggestQuery = {
+    index,
+    _source: false, // disable return of source document info to improve response time
+    suggest: generateSuggestQuery(suggestParameters),
+  };
+
+  const res = await client.search(suggestQuery);
+
+  return res.suggest;
```

Comment:
What does the API return? Is it just a string, the ES suggesters objects or something else?

Reply:
I typed this and reformatted it to a less clunky format (ES loves arrays of things that will only ever have one entry). It looks like this now:

```
{
  taxonomyLevelNameCompletion: [
    { text: 'suggest_0', score: 1 },
    { text: 'suggest_1', score: 1 },
    { text: 'suggest_2', score: 1 },
    { text: 'suggest_3', score: 1 },
    { text: 'suggest_4', score: 1 },
  ];
}
```